File: 0112-path-sum/0112-path-sum.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def hasPathSum(self, root: Optional[TreeNode], targetSum: int) -> bool:
        
        if root is None:
            
            return False
        
        q= deque()
        q.append((root,root.val))
       

        while q:
            cur,result=q.popleft()
            print(result)

            if cur.left is None and cur.right is None:
                if result == targetSum:
                    return True

            
        

            if cur.left:
                q.append((cur.left,result+cur.left.val))
            if cur.right:
                q.append((cur.right,result+cur.right.val))
            
        return False
        

File: 0112-path-sum/0112-path-sum.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} targetSum
 * @return {boolean}
 */
var hasPathSum = function(root, targetSum) {
    if(!root){
        return false
    }

   let que=[[root,root.val]]

   while(que.length>0){
    let [node,sum]=que.shift()

    if(!node.left &&!node.right){
        if(sum ==targetSum){
            return true
        }
    }
if (node.left) que.push([node.left, sum + node.left.val]);
    if (node.right) que.push([node.right, sum + node.right.val]);
   }
   
   return false
   
    
};